### DISCUSSION TITLE
en/latest/workflow-inputs/

### DISCUSSION THREAD

**Discussion by giscus** [argoproj/argo-workflows#12957] [General]
# en/latest/workflow-inputs/

https://argo-workflows.readthedocs.io/en/latest/workflow-inputs/

**Comment by s1dharth-s** [comment]
This might be a stupid question, but what is the difference in using a volume as a means to share files between tasks and using artifacts?

**Reply by static-moonlight** [reply]
I hope I understand your question correctly:

Network-wise, probably not that much. If you would mount a volume directly in a container (workflow task) and read/write data, that data needs to be transferred over the network. If you declare that data as an artifact, it also need to be transferred over the network. So no real big difference here. But I have a case where I could either read a set of json files from an NFS share, or I put them in an artifact, which automatically compresses those files down to <10% of the normal size. Which can have a significant impact to the data volume that goes over the network.

Other than that. it depends a little on the volume you are using to share data and how you use it. For example, how many client are using that share, especially, who deletes data, and when? Simply put: you need to work out how housekeeping is supposed to happen on that volume all by yourself. Lets say that you want to be able to reprocess a failed workflow. You need to keep track of what data needs to be preserved and what data can go (and when).

Argo artifacts kind of does all that for you already. Technically, you could of course use an artifact repository, that is automatically cleaned up every 5 minutes, and practically sabotage Argo's artifact management. But assuming you are not doing that. It's a pretty nice system to have, in terms of workflow isolation and repeatability.

There seems to be a few problems with the artifact garbage collection though, but it's pretty good already ;-)
